### ISSUE TITLE
Account does not have permissions to modify containers

### ISSUE THREAD

**Issue by cyanisaac** [anthonya1999/GoodNight#99]
![Failing](http://i.imgur.com/vyrnv5m.png)

Why does this happen, and how can I fix it?

**Comment by paradoxally** [comment]
Also have this issue. Anyone have a fix?

**Comment by merlinn31** [comment]
I have the same issue.

**Comment by merlinn31** [comment]
Actually, this seems like an XCode versioning issue.  I had the newest version 7.3.  After downloading 7.2 and running through the instructions, it built just fine and is running on my phone now.  

I figured this out all by myself without any help from anyone.

**Comment by cyanisaac** [comment]
Huh, hopefully Xcode fixes that! I have heard that Xcode 7.3 is buggy, I upgraded because I upgraded my iPhone :/.

Any fix for Xcode 7.3?

**Comment by anthonya1999** [comment]
Thanks for pointing this out. I actually have not yet built on Xcode 7.3, since Night Shift is now a feature in iOS 9.3. But I will try to see if I can replicate the issue on my MacBook.

**Comment by sverrevh** [comment]
Same issue here, getting 'Communication with Apple failed' error as well. I think GoodNight works better than NightShift ;) (NightShift doesn't have the fading effect, it's on or off)

**Comment by yanowitz** [comment]
Night shift is not as good as GoodNight, unfortunately. Lacks dark room mode among other things. I too am having this XCode issue.

**Comment by tonytien** [comment]
I'm also having this issue. A quick search for "Communication with Apple failed" message brought up a thread on Stack Overflow about how there may be maintenance going on right now, but system status shows me that everything is fine right now.

**Comment by greatquux** [comment]
Another person having this issue with Xcode 7.3. I am hoping to use GoodNight with my iPhone 5 since it doesn't support Night Shift (requires 64-bit) and f.lux doesn't restore the screen in the morning for me (and they won't support it since Apple gave them crap).

**Comment by gcr** [comment]
I'm getting this error too.

Building with app groups turned off doesn't work because the app tries to load the gamma table from a file with a path relative to the app group container folder, so it's not surprising.

**Comment by paradoxally** [comment]
So I take it no one has managed to compile the app using 7.3? If not, that's a real shame because it can't be deployed on iOS 9.3 devices that can't use Night Shift.

**Comment by sverrevh** [comment]
Looks like Apple found *some how* a way to block this kind of side loading. Maybe it checks for 'goodnight' in the code or name(s)

**Comment by paradoxally** [comment]
Perhaps a solution would be to ignore compiling the widget and instead of loading the gamma table from the app group container, have the app load the table from a file or in code instead. That way it could be compiled as a standalone app without the need for app groups which are troublesome now.

**Comment by nikissk** [comment]
I manage to get it work that way: Download xcode 7.2 from kickass, start it when download complete and load the app. Add App Groups successfully. Next i connect my iPhone 5 ios 9.3. Compile the app was unsuccessful due newest IOS version. Then I close xcode 7.2 and start 7.3. Then App Groups start work fine. Build project and upload it on my phone and still works fine.

**Comment by greatquux** [comment]
Yep, I can confirm that adding the "app groups containers" entitlement does work with Xcode 7.2.1.  I downloaded the DMG file from Apple (https://developer.apple.com/downloads/) and opened it up, and ran the old version right from the DMG (though it had to install some other dependencies), opened up the GoodNight project, fixed the issue (doing it in both the main app and the widget though I probably only had to do it once), closing the old version, and opening up the project in Xcode 7.3, and then I was able to build and run the app on my iPhone 5.  Hopefully Apple will fix the bug in Xcode at some point.

**Comment by mariokorte** [comment]
Hm, I cannot reproduce this with Xcode 7.3. I can manage the AppGroups with that preference pane and run the app just fine. Have you configured your developer accounts (AppleID) correctly in Xcode preferences?

**Comment by greatquux** [comment]
I don't think you would be able to reproduce it if your Apple ID already had the "app groups containers" permission. And I don't think it's a problem with GoodNight itself either, but rather an Xcode bug in 9.3 that doesn't allow 9.3 to properly add the permission for this to the Apple ID.  Xcode 9.2.1 adds the permission to my Apple ID fine and then 9.3 compiles and installs GoodNight properly (which is running and working beautifully).  So I think you can probably close this out and maybe add a note to the README about it, and hopefully Apple will fix it soon.

**Comment by eyv** [comment]
Confirmed: nikissk's and greatquux' instructions WFM. Thanks everyone!
BTW, you don't even have to close the second instance of Xcode -- you can run both (one from the DMG, and the other from Applications) at the same time and even edit the same project at the same time. Not sure how this doesn't munge the files being simultaneously edited, but hey, it works!

**Comment by alyssalng** [comment]
Were you guys able to successfully install Xcode 7.2? I tried to, but my computer got stuck and said there was an error reading the code, so Xcode.app didn't finish copying to the applications. Like a lot of you, I'm on a phone that doesn't have Night Shift because it's too old. 

Any tips would be awesome.

**Comment by Kkevsterrr** [comment]
I'm going to bump this - any fix or update regarding the bug? I'm having the same issue as everyone else in the thread, and any help would be much appreciated!

**Comment by eyv** [comment]
The workaround no longer works for building for iOS 10 using Xcode-beta 8

**Comment by PengQ1** [comment]
I face the same issue now, my device is ios 13 beta, and my Xcode is 11.0 beta, does anyone know how to fix it?

**Comment by nir-jackson-papaya** [comment]
@PengQ1 
Had the same issue, this is what solved it for me:
 if you have "Merchant IDs" simply uncheck them, wait a few seconds and then recheck them.
